pkgs/applications/misc/candle/candle-settings.diff

```diff
@@ -0,0 +1,7 @@
+--- a/frmmain.cpp	1969-12-31 19:00:01.000000000 -0500
++++ b/frmmain.cpp	2021-08-16 22:17:17.502429125 -0400
+@@ -88,1 +88,3 @@
+-    m_settingsFileName = qApp->applicationDirPath() + "/settings.ini";
```

Comment:
Isn't this a bug in the application? If you ran this on any other Linux distro wouldn't it try to write a directory the user doesn't have access to write to? 🤔

Reply:
I take your point... but I've used Candle on several machines (Windows, and various Debian derivatives), and never had to _think_ about where the settings file was, the software just worked as expected. I have to assume they are patching it downstream.

An upstream fix seems unlikely -- the [git repo](https://github.com/Denvi/Candle/) has 285 open issues, and 18 open pull requests dating back a few years.  It's handy software for people driving CNC's, it's a shame that it isn't getting more attention.

Reply:
Fair. It makes sense this would work on Windows, depending on how the application was installed and ran. Any chance you can look into what those other distros do here? No worries if that sounds like too much of a hassle... :smile:

I think what you wanted to do was something like this:
```cxx
// follow XDG standards
m_settingsFileName = getenv("XDG_CONFIG_HOME") ? std::string(getenv("XDG_CONFIG_HOME")) : (getenv("HOME") ? (std::string(getenv("HOME")) + "/.config") : std::string("/tmp"));
m_settingsFileName.append("/candle/settings.ini");
```

Pardon my sloppy `c++`.

You should also:
- Drop the `qDebug`, which I assume you had in there for testing.
- Include a comment which explains the situation in full.
- ping @matti-kariluoma as the maintainer to make sure they're good with this change, or have any insight on the problem.

Thanks for contributing this @gmfawcett :tada:

Reply:
Thanks very much for the feedback.  No worries about your C++, mine is just as sloppy. Note that these are QStrings (Qt), not std::strings -- but I can work in the XDG_CONFIG_HOME variant, I forgot about that one. :)

I suggest that we leave the debug statement. The application is already debug-noisy, and this particular piece of information (where is my config being stored) could be useful to the user, esp. given that this is a downstream patch. But I don't feel strongly about it, I will remove it if that's still your preference.

I will be offline for the next few days but will follow up next week on the actions you suggested. Thanks again for your review. :)

Reply:
That all sounds great :+1:

Reply:
My hunch would be that something like

```c++
m_settingsFileName = QStandardPaths::locate(QStandardPaths::ConfigLocation, "settings.ini");
if (m_settingsFileName.isEmpty()) {
  m_settingsFileName = QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + "/settings.ini";
}
```

would be better... but since being properly cross-platform might not be a priority here so I'm not too fussed. It would be nice if we could have it work properly on Darwin, though, I guess.

Reply:
It's been years since I've written any Qt code and I forgot about that one! Good catch.

Reply:
@lukegb  thanks! I've updated the PR to use that approach.  Works well for me on Linux, I can't test on Darwin unfortunately.